rrtmgp/kernels-openacc/mo_gas_optics_kernels.F90

```diff
@@ -599,10 +672,12 @@ subroutine compute_Planck_source(                        &
     ! Map to g-points
     !
     !$acc parallel loop collapse(3)
-    do igpt = 1, ngpt
+    do icol = 1, ncol, 2
```

Comment:
It seems like you're explicitly unrolling a loop here - why?

Reply:
Added comments. Increased number of memory transactions help to improve memory bandwidth quite a bit, and this loop is pretty expensive

Reply:
I don't quite understand this answer. Does it mean that you're explicitly telling the compiler that it can work on two sets of calculations at once? Isn't that obvious from the loop collapse?

Reply:
On the GPU typically every loop iteration will be mapped to a single CUDA thread. Only a limited number of threads can be active at a time, just like in the CPU. What I did, is mapped two iteration from different memory locations onto single thread, so that each of the threads can issue two memory instructions. This allows more memory transactions in flight on the GPU, and therefore higher bandwidth (about 15-20% higher in that case). The general thing is called Instruction-Level Parallelism https://en.wikipedia.org/wiki/Instruction-level_parallelism

Reply:
Thanks, that's interesting. I'm only surprised that the compiler can't make (or be told to make) this kind of optimization without restructuring code.